src/main/scala/tech/cryptonomic/conseil/routes/Tezos.scala

```diff
@@ -62,49 +61,74 @@ object Tezos {
 class Tezos(implicit apiExecutionContext: ExecutionContext) extends LazyLogging with DatabaseApiFiltering {
 
   import Tezos._
+  import JsonUtil.{toJson, JsonString}
 
   /* reuse the same context as the one for ApiOperations calls
    * as long as it doesn't create issues or performance degradation
    */
   override val asyncApiFiltersExecutionContext = apiExecutionContext
 
-  //this automatically accepts any type `T` as content for calling [[RequestContext.complete]]
-  //converts to json string via JsonUtil adding the correct content-type to the response entity
-  implicit def jsonStringMarshaller[T]: ToEntityMarshaller[T] =
+  //add the correct content-type for [[JsonUtil]]-converted values
+  implicit val jsonMarshaller: ToEntityMarshaller[JsonString] =
     PredefinedToEntityMarshallers.StringMarshaller
-      .compose(JsonUtil.toJson[T])
-      .wrap(MediaTypes.`application/json`)(identity)
+      .compose((_: JsonString).json)
+      .wrap(MediaTypes.`application/json`)(identity _)
+
+  /* allow generic handling of optional results, embedded in async computations
```

Comment:
Did you type all of this out yourself? If you type "/**<enter>" in IntelliJ, it automatically handles parameter docs. I'd suggest adding that extra start on the top part for proper JavaDoc syntax.

Reply:
Oh, yes, sorry. I'm doing it manually.

Actually I've seen that javadoc syntax is used in the codebase, but scaladoc syntax have different conventions in this regard: https://docs.scala-lang.org/style/scaladoc.html

That's why I might be making mistakes at times